src/main/java/org/dependencytrack/parser/cyclonedx/util/ModelConverter.java

```diff
@@ -84,15 +84,32 @@ private ModelConverter() { }
      */
     public static List<Component> convertComponents(final QueryManager qm, final Bom bom, final Project project) {
         final List<Component> components = new ArrayList<>();
-        if (bom.getComponents() != null) {
-            for (int i = 0; i < bom.getComponents().size(); i++) {
-                final org.cyclonedx.model.Component cycloneDxComponent = bom.getComponents().get(i);
-                if (cycloneDxComponent != null) {
-                    components.add(convert(qm, cycloneDxComponent, project));
+        final List<String> purls = new ArrayList<>();
+        components.addAll(convertComponents(qm, bom.getComponents(), project, purls));
+        if (bom.getMetadata() != null && bom.getMetadata().getComponent() != null) {
+            components.addAll(convertComponents(qm, bom.getMetadata().getComponent().getComponents(), project, purls));
+        }
+        return components;
+    }
+
+    private static List<Component> convertComponents(final QueryManager qm, final List<org.cyclonedx.model.Component> components,
+            final Project project, final List<String> purls) {
+        final List<Component> convertedComponents = new ArrayList<>();
+        if (components != null) {
+            for (org.cyclonedx.model.Component component : components) {
+                if (component != null) {
+                    boolean hasPurl = component.getPurl() != null && !component.getPurl().isBlank();
+                    if (hasPurl && purls.contains(component.getPurl())) {
```

Comment:
Since purl is also optional, we might also need bom-ref and CPE-based checks. I suggest waiting for a response from DT developers before making this change.